### DISCUSSION TITLE
Disable CSS reset via config or other means

### DISCUSSION THREAD

**Discussion by jamiethorpe** [vuejs/vitepress#1518] [Ideas]
I've recently been documenting a UI library using Vitepress and have found myself having to override a number of styles so that my components look the same as in the rest of my applications. I also noticed this only occurred after switching from 0.22.4 to 1.0.0-alpha.21.

It would be extremely useful to be able to either disable the CSS reset via a config property, or even by wrapping sections with a particular tag. Is that something that would be possible?

That aside, it's an awesome tool for documentation.

**Comment by jamiethorpe** [comment]
I'm not 100% certain it's _only,_ the CSS reset, but what lead me to believe it was the reset was that I could add the following styles to get my inputs looking how they did in 0.22.4 and in other applications.

```css
input {
    border: solid;
}
```
Result
![image](https://user-images.githubusercontent.com/6597612/197199616-9cecc0a9-8f7d-4160-961c-fac37eaa513f.png)

Whereas if I add the same for inputs within vp-doc, I get a result that requires a bit more styling to get back to the same result:

```css
.vp-doc input {
    border: solid;
}
```

Result
![image](https://user-images.githubusercontent.com/6597612/197200142-1757a137-7a3f-41e1-bb5b-ac84ea787176.png)

I have found myself needing to target selectors within vp-doc as well though, so it really could be a mix of both.

@brc-dd  also mentioned https://whyframe.dev/docs/integrations/vitepress earlier in Discord, so that could be a solid solution.

**Reply by gorlanova** [reply]
I don't know if you see updates under my own reply, but in case, there's a solution :
https://github.com/vuejs/vitepress/discussions/1518#discussioncomment-7403960

**Comment by gorlanova** [comment]
I came here looking for a fix, and stumbled upon this thread, I'm having the exact same issue, it would be amazing to be able to pass a class or attribute to a wrapper for my lib demos, rather than overriding vitepress styling on every component :)

**Reply by brc-dd** [reply]
Refer the style-isolation part of this - https://vitepress.dev/guide/markdown#raw -- you'll need to add vp-raw class to your lib demos (or any of their parent wrappers). It will disable only the base styles by default. You can configure it to include doc styles as well - `includeFiles: [/vp-doc\.css/, /base\.css/]` (you might need to adjust the file regexes depending on your setup).

**Reply by gorlanova** [reply]
Oh fab ! Thank you SO MUCH for that tip that's going to make things way easier !
And thanks for everything else, Vitepress is such a great tool
